File: map_ts/src/Entity/entity-structure.ts
import type { EntityId } from "./entity-id.js";
import type { PathId } from "../Path/path-id.js";

export type EntityStructure = {
  entity: EntityId;

  incomingPaths: PathId[];
  outgoingPaths: PathId[];

  parent?: EntityId;
  children: EntityId[];
};

File: map_python/src/Entity/entity_structure.py
from dataclasses import dataclass, field

from .entity_id import EntityId
from ..Path.path_id import PathId


@dataclass(slots=True)
class EntityStructure:
    entity: EntityId
    incoming_paths: list[PathId] = field(default_factory=list)
    outgoing_paths: list[PathId] = field(default_factory=list)
    parent: EntityId | None = None
    children: list[EntityId] = field(default_factory=list)
